lib/pem.ml

```diff
@@ -91,7 +94,15 @@ module Cs = struct
     concat "" (first @ lines @ last)
 end
 
-let parse, unparse = Cs.(parse, unparse)
+let parse_with_errors, unparse = Cs.(parse_with_errors, unparse)
+
+let parse data =
+  let rec all_ok acc = function
```

Comment:
can we use some list helper function? filter_map or exists or find_opt or fold?

Reply:
Sure! I went with `List.partition_map` which was introduced in ocaml 4.12, as x509 already requires ocaml >=4.13